aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -358,6 +362,18 @@ Tensor sparse_compressed_to_dense(
   TORCH_CHECK(
       !dtype.has_value(),
       "dtype argument is not supported by sparse_csr_to_dense");
+
+  // Guard upfront against hybrid tensors (causes segfault)
+  auto batch_ndim = AT_DISPATCH_ROW_SPARSE_COMPRESSED_LAYOUTS(
```

Comment:
more information on what causes segfaults?

Reply:
The issue lies in forwarding to `dense.add_(sparse_compressed)` for csr/csc which will segfault when sparse_compressed has dense dims. Likely due to an assumption about the lack of dense dims without an actual check.